lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -46,17 +46,48 @@ class PublicLayoutHelper
       attr_reader :footer_navigation, :footer_meta, :cookie_banner_data
 
       def initialize(local_assigns)
+        @footer_links_count = generate_ga4_footer_links_count(local_assigns)
```

Comment:
We must calculate the amount of footer links as the first thing in this function. This is because the two lines beneath this are where the links are actually generated and have their data attributes added to them. Therefore the amount of links must be calculated here otherwise the data attributes for everything will already be set if we do it at a later time in this function.